AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Storage/SQLite/SQLStatement+Condition.swift

```diff
@@ -76,7 +87,56 @@ struct ConditionStatement: SQLStatement {
                                                        predicate: predicate,
                                                        namespace: namespace)
         self.stringValue = sql
+        let (sql2, variables2) = translateQueryPredicate2(from: modelSchema,
+                                                          predicate: predicate,
+                                                          namespace: namespace)
+        self.stringValue2 = sql2
         self.variables = variables
     }
+}
+extension QueryPredicate {
+    func conditionSQL(namespace: Substring? = nil,
+                      indentPrefix: String = "  ",
+                      indentSize: Int = 0) -> String {
+        if let group = self as? QueryPredicateGroup {
+            return group.conditionSQL(namespace: namespace,
+                                      indentPrefix: indentPrefix,
+                                      indentSize: indentSize)
+        } else if let operation = self as? QueryPredicateOperation {
+            return operation.conditionSQL(namespace: namespace,
+                                          indentPrefix: indentPrefix,
+                                          indentSize: indentSize)
+        } else if let constant = self as? QueryPredicateConstant,
+                  case .all = constant {
+            return "or 1 = 1"
+        }
+        preconditionFailure("Failed \(String(describing: self))")
+    }
+}
+extension QueryPredicateGroup {
+    func conditionSQL(namespace: Substring? = nil,
+                      indentPrefix: String,
+                      indentSize: Int) -> String {
+        let indent = String(repeating: indentPrefix, count: indentSize)
+        let innerIndent = String(repeating: indentPrefix, count: indentSize + 1)
+        let groupType = "\n\(innerIndent)\(type.rawValue) "
+        let conditionSQL = Array(predicates.map { (predicate) -> String in
+            return predicate.conditionSQL(namespace: namespace,
+                                          indentPrefix: indentPrefix,
+                                          indentSize: indentSize + 1)
+        }).joined(separator: groupType)
```

Comment:
shoving the constant "and"/"or" group type in between each predicate (regardless of whether they are an operation or a group)